wrapper/src/main/java/software/amazon/jdbc/ConnectionPluginManager.java

```diff
@@ -498,8 +499,34 @@ public HostSpec getHostSpecByStrategy(HostRole role, String strategy)
 
       throw new UnsupportedOperationException(
           "The driver does not support the requested host selection strategy: " + strategy);
-    } catch (RuntimeException e) {
-      throw e;
+    } catch (Exception e) {
+      throw new SQLException(e);
+    }
+  }
+
+  public HostSpec getHostSpecByStrategy(List<HostSpec> hosts, HostRole role, String strategy)
```

Comment:
This method is a copy of original method. We need to reuse the code here.

Reply:
Done!